src/storage/core.rs

```diff
@@ -888,9 +924,11 @@ where
     /// Mark as deleted entries with matching key
     /// # Errors
     /// Fails after any disk IO errors.
-    pub async fn mark_all_as_deleted(&self, key: impl AsRef<K>) -> Result<u64> {
+    pub async fn mark_all_as_deleted(&self, key: impl AsRef<K>, force_write: bool) -> Result<u64> {
```

Comment:
I don't like `mark_all_as_deleted` name for exported function. Maybe, it is better to rename it to `delete`?

Also, the new argument has a bad name too. I suggest inverting its meaning and changing its name to `only_if_presented`